pkg/sdk/testint/external_functions_integration_test.go

```diff
@@ -18,53 +17,52 @@ func TestInt_ExternalFunctions(t *testing.T) {
 	integration, integrationCleanup := testClientHelper().ApiIntegration.CreateApiIntegration(t)
 	t.Cleanup(integrationCleanup)
 
-	cleanupExternalFunctionHandle := func(id sdk.SchemaObjectIdentifier, dts []sdk.DataType) func() {
+	cleanupExternalFunctionHandle := func(id sdk.SchemaObjectIdentifierWithArguments) func() {
 		return func() {
-			err := client.Functions.Drop(ctx, sdk.NewDropFunctionRequest(sdk.NewSchemaObjectIdentifierWithArguments(id.DatabaseName(), id.SchemaName(), id.Name(), dts...)).WithIfExists(true))
+			err := client.Functions.Drop(ctx, sdk.NewDropFunctionRequest(id).WithIfExists(true))
 			require.NoError(t, err)
 		}
 	}
 
-	// TODO [SNOW-999049]: id returned on purpose; address during identifiers rework
-	createExternalFunction := func(t *testing.T) (*sdk.ExternalFunction, sdk.SchemaObjectIdentifier) {
+	createExternalFunction := func(t *testing.T) *sdk.ExternalFunction {
 		t.Helper()
-		id := testClientHelper().Ids.RandomSchemaObjectIdentifierWithArgumentsOld(defaultDataTypes...)
+		id := testClientHelper().Ids.RandomSchemaObjectIdentifierWithArguments(defaultDataTypes[0])
 		argument := sdk.NewExternalFunctionArgumentRequest("x", defaultDataTypes[0])
 		as := "https://xyz.execute-api.us-west-2.amazonaws.com/production/remote_echo"
-		request := sdk.NewCreateExternalFunctionRequest(id, sdk.DataTypeVariant, sdk.Pointer(integration.ID()), as).
-			WithOrReplace(sdk.Bool(true)).
-			WithSecure(sdk.Bool(true)).
+		request := sdk.NewCreateExternalFunctionRequest(id.SchemaObjectId(), sdk.DataTypeVariant, sdk.Pointer(integration.ID()), as).
+			WithOrReplace(true).
+			WithSecure(true).
 			WithArguments([]sdk.ExternalFunctionArgumentRequest{*argument})
 		err := client.ExternalFunctions.Create(ctx, request)
 		require.NoError(t, err)
-		t.Cleanup(cleanupExternalFunctionHandle(id.WithoutArguments(), []sdk.DataType{sdk.DataTypeVariant}))
+		t.Cleanup(cleanupExternalFunctionHandle(id))
 
 		e, err := client.ExternalFunctions.ShowByID(ctx, id)
 		require.NoError(t, err)
-		return e, id
+		return e
 	}
 
-	assertExternalFunction := func(t *testing.T, id sdk.SchemaObjectIdentifier, secure bool) {
+	assertExternalFunction := func(t *testing.T, id sdk.SchemaObjectIdentifierWithArguments, secure bool) {
 		t.Helper()
-		dts := id.Arguments()
 
 		e, err := client.ExternalFunctions.ShowByID(ctx, id)
```

Comment:
we could add an id equality check (instead of checking name, schema name and arguments separately)

Reply:
True, changed